src/main.rs

```diff
@@ -1283,13 +1310,57 @@ impl PofToolsGui {
                         }
                     }
 
-                    let size = 0.05 * model.sub_objects[obj_id].radius;
+                    let radius = model.sub_objects[obj_id].radius;
+                    let size = 0.05 * radius;
+                    let pos = model.get_total_subobj_offset(obj_id);
 
                     let mut lollipop_origin = GlLollipopsBuilder::new(LOLLIPOP_SELECTED_POINT_COLOR);
-                    lollipop_origin.push(model.get_total_subobj_offset(obj_id), Vec3d::ZERO, size);
+                    // Origin lollipop
+                    lollipop_origin.push(pos, Vec3d::ZERO, size);
                     let lollipop_origin = lollipop_origin.finish(display);
-
                     self.lollipops = vec![lollipop_origin];
+
+                    match model.sub_objects[obj_id].uvec_fvec() {
+                        Some((uvec, fvec)) => {
+                            // Set up arrowhead sticks
+                            let stick_length = 2. * radius;
+                            // Blue lollipop (stick only) for uvec
+                            let mut lollipop_uvec = GlLollipopsBuilder::new(LOLLIPOP_SELECTED_BANK_COLOR);
```

Comment:
Although you do specify stick only here, can you rename the locals `stick_...` instead of `lollipop_...` where possible, just to be a little clearer?